docs/source/backends/zerotier.rst

```diff
@@ -325,8 +325,11 @@ Example:
         "id": ["9536600adf654321"],
         "name": "ow_zt",
     }
+    nw_id = server_config['id'][0]
     client_config = OpenWrt.zerotier_auto_client(
-        nwid=server_config['id'], name=server_config['name']
+        nwid=server_config['id'],
+        name=server_config['name'],
+        ifname=f'owzt{nw_id[-6:]}',
```

Comment:
BTW, why don't we add a schema property for this?
If the property is present we can automatically generate the other config options as you're doing here, with the difference that these values will be hidden to the user in the template, they will just see "ifname". Wouldn't this be better?
Let me know if my request is clear.

Reply:
Certainly, we can enhance the user experience by introducing an `ifname` property to the schema while simplifying the removal of other advanced configuration options.

Here's my proposed approach:

1. To accommodate the custom ZeroTier interface names associated with each ZeroTier network a device belongs to, we can modify **schema.py**. Specifically, we can add the `nwid_ifname` property, which will be an array of objects containing `nwid` and `ifname` for each ZeroTier network. Additionally, we can ensure that the `maxLength` property is set to `16` for `nwid` and `10` for `ifname`, making both properties required for this object.

2. In accordance with your above review suggestion, we can display only the main configuration options in the templates while concealing the advanced ones.

**Visible options:**
- `name`
- `nwid_ifname`
- `identity_secret`
- `config_path` (This can be customized by users in VPN client template)

**Hidden options:**
- `config_path`
- `copy_config_path`
- ZeroTier interface file object located at **/etc/ow_zerotier_extra/devicemap**

3. To implement the above changes, it will be necessary to update the code in `openwrt/converters/zerotier.py`.

### RESULTS:

#### With single `nwid_ifname` object

![Screenshot from 2023-08-21 16-18-12](https://github.com/openwisp/netjsonconfig/assets/56113566/01e24f89-9832-417a-be7a-d39753c3a647)

![Screenshot from 2023-08-21 16-18-27](https://github.com/openwisp/netjsonconfig/assets/56113566/0f080d53-ea68-460b-893d-2111612fab47)

#### With multiple `nwid_ifname` objects

![Screenshot from 2023-08-21 16-24-55](https://github.com/openwisp/netjsonconfig/assets/56113566/26cbf14d-af75-4c50-a39d-01df0943b3e6)

![Screenshot from 2023-08-21 16-19-12](https://github.com/openwisp/netjsonconfig/assets/56113566/7d1d4ef1-0cae-45ac-845f-006ff7ec8423)